### ISSUE TITLE
Can laugh and other emotions transfer from one language to other?

### ISSUE THREAD

**Issue by cod3r0k** [stlohrey/dia-finetuning#11]
"Hi, I think that when I fine-tune it in another language, the original key features in English are forgotten. Why does this happen?"

**Comment by stlohrey** [comment]
You have to include these features in your finetuning. You can check if they transfer between languages, but if you want to keep the features from the english model (or even english language), you will have to include that in your finetuning dataset.

**Comment by cod3r0k** [comment]
Thanks! But do you have any guidance on how to add emotion labels?
@stlohrey, do you happen to have any sample guidelines for the recording or tagging process of an audio TTS emotion dataset?
The dataset you mentioned, `Paradoxia/opendata-iisys-hui`, also doesn't seem to include any emotion labels.

**Comment by jeanca95** [comment]
I will think given the transformer structure and that is a byte-level tokenization. My hypothesis will be that if you include non-verbal sounds in your dataset and in your csv transcription you include the tags as (coughs), (clears throat) etc. The model should be able to learn this sounds. I will test this approach in the future. Where you able to train in another language? I'm wondering how did you tag your dataset and how did you perform inference on the model. Given the dataset I didn't include tags like [S1] or [S2] as used in the original. The inference text in this case will be using: text = "[language tag] 'audio_transcript' + 'text_to_generate'". And it should work. Or am I mistaken? Or how did you tag your audios? Thank you for your help.

**Comment by stlohrey** [comment]
Including non-verbal tags like that should be working.
For inference, you can use text = "[language tag] 'audio_transcript' + 'text_to_generate'" or text = "[language tag] 'audio_transcript' + [language tag]'text_to_generate'", it works both ways, you can experiment with that and see if the latter yields better results.

**Comment by jeanca95** [comment]
What about the [S1] and [S2] tags? Should they be included too in the training dataset transcription. for example like this:
`audio_000000.wav|[es] [S1] Los corazones de pollo son una delicia.`
`audio_000001.wav|[es] [S1] Es un plato muy nutritivo.`

**Comment by stlohrey** [comment]
You would only need do include [S1] and [S2] tags if you want to finetune on a dialogue dataset.

**Comment by Ranjit246** [comment]
Hi, I had a follow-up regarding emotion transfer in fine-tuning:

To effectively transfer emotion during fine-tuning, I understand that non-verbal tags - usually defined in English - should be included in the dataset. My current dataset does not contain these non-verbal tags. Is it valid to manually insert them into the transcripts (where appropriate) and proceed with fine-tuning?

Also, is there any guideline or recommendation for:
 * How frequently these tags should appear in the dataset?
 * Where (semantically or acoustically) they should be placed? Like is just placing at the beginning or ending is fine?

My goal is to fine-tune on a new language, but with improved expressivity using these non-verbal cues. Is this approach compatible with the training strategy discussed here?

Thanks for the great work on this repo - just wanted to ensure I’m aligning with best practices for emotional nuance. @stlohrey

**Comment by stlohrey** [comment]
> Hi, I had a follow-up regarding emotion transfer in fine-tuning:
> 
> To effectively transfer emotion during fine-tuning, I understand that non-verbal tags - usually defined in English - should be included in the dataset. My current dataset does not contain these non-verbal tags. Is it valid to manually insert them into the transcripts (where appropriate) and proceed with fine-tuning?
> 
> Also, is there any guideline or recommendation for:
> 
> * How frequently these tags should appear in the dataset?
> * Where (semantically or acoustically) they should be placed? Like is just placing at the beginning or ending is fine?
> 
> My goal is to fine-tune on a new language, but with improved expressivity using these non-verbal cues. Is this approach compatible with the training strategy discussed here?
> 
> Thanks for the great work on this repo - just wanted to ensure I’m aligning with best practices for emotional nuance. 

Do you mean non verbal sounds like laughing or do you mean global emotion tags? I did experiment with some kind of tonality tag in the beginning of the sequence, and it seemed to work, but would need further experiments to confirm, maybe also special bytes defined for that would help. As for non-verbal sounds, I did not try this, but they should be placed wherever they occur.
If you want to use emotion or tonality tags, I think it might be better to place them in the beginning.
